src/test/java/htsjdk/variant/variantcontext/GenotypeLikelihoodsUnitTest.java

```diff
@@ -125,16 +125,36 @@ public void testGetAsMap(){
     }
 
     @Test
-    public void testCalculateNumLikelihoods() {    
-        
+    public void testCalculateNumLikelihoods() {
+
         for (int nAlleles=2; nAlleles<=5; nAlleles++)
-            // simplest case: diploid
+        {
             Assert.assertEquals(GenotypeLikelihoods.numLikelihoods(nAlleles, 2), nAlleles*(nAlleles+1)/2);
+        }
 
         // some special cases: ploidy = 20, #alleles = 4
         Assert.assertEquals(GenotypeLikelihoods.numLikelihoods(4, 20), 1771);
+        // ploidy = 10, alleles = 5
+        Assert.assertEquals(GenotypeLikelihoods.numLikelihoods(5, 10), 1001);
+        // ploidy = 16, alleles = 10
+        Assert.assertEquals(GenotypeLikelihoods.numLikelihoods(10, 16), 2042975);
+    }
+
+    @Test
+    public void testNumLikelihoodsCache() {
+        GenotypeNumLikelihoodsCache cache = new GenotypeNumLikelihoodsCache();
+
+        cache.put(2, 5, 10);
+        cache.put(10, 16, 2042975);
+
+        Assert.assertEquals(cache.get(2, 5), new Integer(10));
```

Comment:
not sure why we need to manually box this.

Reply:
We shouldn't need to.  I'm not actually sure why I added that